docs/guides/modding-guide.rst

```diff
@@ -0,0 +1,74 @@
+DFHack modding guide
+====================
+
+Preface
+-------
+
+Why not just use raw modding?
+
+It's much less hacky for many things, easier to maintain, and easier to extend. Of course if you're adding a new creature or whatever do use raw modding, but for things like adding syndromes when certain actions are performed, absolutely try out DFHack. Many things will require a mix of raw modding and DFHack.
+
+What kind of things can we make?
+
+Lots of things, and the list grows as more and more hooks and tools are developed within DFHack. You can modify behaviours by cleverly watching and modifying certain areas of the game with functions that run on every tick or by using hooks. Familiarising yourself with the many structs oof the game will help with ideas immensely, and you can always ask for help in the right places (e.g. DFHack's Discord).
+
+You will need to know Lua for the scripting guides herein.
+
+A script-maker's environment
+----------------------------
+
+A script is run by writing its path and name from a script path folder without the file extension. E.g. gui/gm-editor for hack/scripts/gui/gm-editor.lua. You can make all your scripts in hack/scripts/, but this is not recommended as it makes things much harder to maintain each update. It's recommended to make a folder with a name like "own-scripts" and add it to dfhack-config/script-paths.txt. You should also make a folder for external installed scripts from the internet that are not in hack/scripts/.
```

Comment:
This is generally good advice for in-development scripts, but perhaps you should also explain the structure of an installed mod, with supporting scripts in `raw/scripts` and "autorun" scripts in `raw/init.d`. That might help put the rest of this in context. Otherwise, we're just describing how to write a script, not a "mod". Maybe an "Anatomy of a mod" section before this one.

It seems like it would be a best practice to have `init.d` scripts just call other scripts so it's easy to dynamically change behavior (since `init.d` scripts aren't overridden by similarly-named scripts in your dev directory). I.e. `init.d/dustkicker.lua` would just call `dustkickerctl start`, which is run from `raw/scripts` on an end user machine or from own-scripts in your dev env. Actual modders might have other, better advice, though.

It's a *particularly* bad idea to put in-development scripts in `hack/scripts` when writing a mod, since as soon as you test installing your mod in `raw`, any further changes you think you're making to your copy in `hack/scripts` will never be noticed.

Maybe also mention that you should prepend your "own-scripts" path with a `+` so it takes precedence over the default script directories.

Reply:
What do you want pulled out of "modmaker's environment" and into "anatomy of a script"? I feel like they're too similar

Reply:
a script..doesn't have much anatomy, does it? unless we're talking about different specific use cases, like how to use `eventful`, how to do timed scans with `repeat-util`, etc.

Reply:
Sorry I meant what do you want moved into anatomy of a mod